press/api/callbacks.py

```diff
@@ -90,10 +94,9 @@ def callback(job_id: str):
 	if not server:
 		frappe.throw("Not permitted", frappe.ValidationError)
 
-	job = verify_job_id(server, job_id=job_id)
+	job = verify_job_id(server, job_id)
 	if not job:
 		frappe.throw("Invalid Job Id", frappe.ValidationError)
 
-	handle_job_updates(server=server, job_id=job_id)
-	frappe.set_user("guest")
-	return "Success"
+	frappe.enqueue(handle_job_updates, server=server, job_identifier=job_id)
+	frappe.set_user("Guest")
```

Comment:
Maybe reset user in a final block of try..catch

Something like this > https://github.com/frappe/press/blob/693cd5f8e124318aef7c9786749a91d4871a7e0f/press/api/saas.py#L57-L89

Reply:
Does frappe.enqueue raise?

Reply:
Not in usual scenario,  but enqueue also can raise error in certain condition 

Also there is verify_job_id function which do db call that can fail

Also before raising the error by frappe.throw you should unset the session user